docs: user guide for publishing via Modal (#551)
Garden-AI/garden: docs/user_guide/modal-publishing.md

## 1. Create a Modal Account

> **NOTE:** This step requires a free [GitHub](https://github.com) account to authenticate with Modal.

Reviewer: ```suggestion
**Prerequisite:** This step requires a free [GitHub](https://github.com/join) account to authenticate with Modal.
```

> Create a free Modal account: [signup](https://modal.com/signup)

Reviewer: I think we should mention the $30/month of free modal compute 
```suggestion
Create a free Modal account by linking your GitHub: [signup](https://modal.com/signup). 

**NOTE:** You'll also get $30/month of free compute credit for your personal account. It's totally safe to spend those credits developing and debugging your modal functions before you publish them with garden -- modal functions run through garden won't charge your personal account. 
```

## 2. Author a Modal App

### Create a Modal file

Modal files are regular python scripts. Create a new python file in your project:

```bash
$ touch my_app.py
```

> 

Reviewer: ```suggestion

You can also use modal's [interactive playground](https://modal.com/playground/get_started) to develop your modal function.
```

### Imports

```python
# my_app.py

# Import the Modal SDK
import modal
```

### Define the Modal App

> Create a `modal.App` obeject where your custom functions will be registered.

Reviewer: ```suggestion
Create a `modal.App` object where your custom functions will be registered.
```

> Modal apps need to be assigned to a variable named `app` and in the global scope.

Reviewer: ```suggestion
Modal apps need to be assigned to a variable named `app` and at the top-level (global) scope.
```

```
app = modal.App("my-cool-app")
```

> 

Reviewer: I think we need a section on images too so people can specify requirements -- it can be brief but it should link out to the modal docs/playground https://modal.com/playground/custom_container

### Define Modal functions

Modal functions are regular python functions that have been decorated with `@app.function()`.
The `@app.function()` decorator registers the function with the `modal.App` created above.
Like Modal Apps, Modal functions need to be defined in the global scope.

```python
# Define a function and register it with the app
# Functions can be named anything you like
@app.function()
def my_awesome_function(data):
  result = sum(data)
  return result

# You can register multiple functions to the same app
@app.function()
def my_other_cool_function(data):
  # you can call modal functions from within other modal functions

>   inital_results = my_awesome_function.remote(data)

Reviewer: I think we should cut this example, we don't want our users trying to call modal functions from inside other modal functions since that would make it easy to dodge usage limits

Author: Good catch! I didn't think of that!

  result = inital_results + 42
  return result

```

### The Complete Modal File

That is all we need to do to define a Modal App that can be published with Garden! See the complete modal file below:
See the [Modal docs](https://modal.com/docs/guide) for more details on defining Modal Apps and functions.

```python
# my_app.py

# Import the Modal SDK
import modal

# Define your App -- this is the top-level entity that holds references to functions
# It must be assigned to a variable named 'app' for Garden to extact it properly
app = modal.App("my-cool-app")

# Define a function and register it with the app
# Functions can be named anything you like
@app.function()
def my_awesome_function(data):
  result = sum(data)
  return result

> # You can register multiple functions to the same app

Reviewer: I like that the example shows off multiple functions, so maybe just cut the part where one function calls the other one

@app.function()
def my_other_cool_function(data):
  # you can call modal functions from within other modal functions
  inital_results = my_awesome_function.remote(data)
  result = inital_results + 42
  return result
```

## 3. Upload your App to Garden

> ### Create a new Garden

Reviewer: really love the screenshots here
